File: app/src/main/java/cn/yuntk/radio/ibook/receiver/NoisyAudioStreamReceiver.java
package cn.yuntk.radio.ibook.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;


import cn.yuntk.radio.ibook.service.AudioPlayer;

/**
 * 来电/耳机拔出时暂停播放
 * Created by wcy on 2016/1/23.
 */
public class NoisyAudioStreamReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        AudioPlayer.get().playPause();
    }
}

File: app/src/main/java/cn/yuntk/radio/play/NoisyAudioStreamReceiver.kt
package cn.yuntk.radio.play

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import cn.yuntk.radio.Constants
import cn.yuntk.radio.utils.LogUtils

/**
 *  来电/耳机拔出时暂停播放
 * Author : Gupingping
 * Date : 2018/7/16
 * Mail : gu12pp@163.com
 */
class NoisyAudioStreamReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
//        PlayService.startCommand(context, ACTION_MEDIA_PLAY_PAUSE)
        if (intent.action == Constants.HEADSET_PLUG) {
            if (intent.hasExtra("state")) {
                when (intent.getIntExtra("state", 0)) {
                    0 -> {
                        LogUtils.e("headset not connected")
                        PlayManager.instance.playPause()
                    }
                    1 -> {
                        if (!PlayManager.instance.isPlaying()) {
                            PlayManager.instance.playPause()
                            LogUtils.e(" headset  connected")
                        }
                    }
                }
            }
        }
    }
}